test/extended/etcd/helpers/helpers.go

```diff
@@ -184,37 +184,36 @@ func recoverClusterToInitialStateIfNeeded(ctx context.Context, t TestingT, machi
 }
 
 func DeleteSingleMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (string, error) {
-	waitPollInterval := 15 * time.Second
+	waitPollInterval := 30 * time.Second
 	waitPollTimeout := 5 * time.Minute
-	t.Logf("Waiting up to %s to delete a machine", waitPollTimeout.String())
-
 	machineToDelete := ""
-	err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
-		machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
-		if err != nil {
-			return isTransientAPIError(t, err)
-		}
 
-		// Machine names are suffixed with an index number (e.g "ci-op-xlbdrkvl-6a467-qcbkh-master-0")
-		// so we sort to pick the lowest index, e.g master-0 in this example
-		machineNames := []string{}
-		for _, m := range machineList.Items {
-			machineNames = append(machineNames, m.Name)
-		}
-		sort.Strings(machineNames)
-		machineToDelete = machineNames[0]
-		t.Logf("attempting to delete machine %q", machineToDelete)
+	machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
+	if err != nil {
+		return "", fmt.Errorf("error listing master machines: '%w'", err)
+	}
+
+	// Machine names are suffixed with an index number (e.g "ci-op-xlbdrkvl-6a467-qcbkh-master-0")
+	// so we sort to pick the lowest index, e.g master-0 in this example
+	machineNames := []string{}
+	for _, m := range machineList.Items {
+		machineNames = append(machineNames, m.Name)
+	}
+	sort.Strings(machineNames)
+	machineToDelete = machineNames[0]
+
+	t.Logf("Waiting up to %s to delete a machine", waitPollTimeout.String())
 
+	err = wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		t.Logf("attempting to delete machine '%q'", machineToDelete)
 		if err := machineClient.Delete(ctx, machineToDelete, metav1.DeleteOptions{}); err != nil {
-			// The machine we just listed should be present but if not, error out
 			if apierrors.IsNotFound(err) {
-				t.Logf("machine %q was listed but not found or already deleted", machineToDelete)
-				return false, fmt.Errorf("machine %q was listed but not found or already deleted", machineToDelete)
+				t.Logf("machine '%q' was listed but not found or already deleted", machineToDelete)
+				return false, nil
 			}
 			return isTransientAPIError(t, err)
```

Comment:
We'll can leave off detecting transient API errors here as the client should retry and we shouldn't need to handle those per https://github.com/openshift/origin/pull/27702#discussion_r1119384119
```suggestion
			return false, err
```

Reply:
done